diff --git a/src/common/peer.cpp b/src/common/peer.cpp
index 3785326..876edcb 100644
--- a/src/common/peer.cpp
+++ b/src/common/peer.cpp
@@ -19,6 +19,7 @@
  ***************************************************************************/
 
 #include "peer.h"
+#include <QtCore/QDataStream>
 
 Peer::Peer(AuthHandler *authHandler, QObject *parent)
     : QObject(parent)
diff --git a/src/client/treemodel.cpp b/src/client/treemodel.cpp
index f599803..952c778 100644
--- a/src/client/treemodel.cpp
+++ b/src/client/treemodel.cpp
@@ -556,9 +556,9 @@ void TreeModel::endAppendChilds()
     ChildStatus cs = _childStatus;
 #ifndef QT_NO_DEBUG
     QModelIndex parent = indexByItem(parentItem);
-#endif
     Q_ASSERT(cs.parent == parent);
     Q_ASSERT(rowCount(parent) == cs.childCount + cs.end - cs.start + 1);
+#endif
 
     _aboutToRemoveOrInsert = false;
     for (int i = cs.start; i <= cs.end; i++) {
@@ -605,9 +605,9 @@ void TreeModel::endRemoveChilds()
 #ifndef QT_NO_DEBUG
     ChildStatus cs = _childStatus;
     QModelIndex parent = indexByItem(parentItem);
-#endif
     Q_ASSERT(cs.parent == parent);
     Q_ASSERT(rowCount(parent) == cs.childCount - cs.end + cs.start - 1);
+#endif
     _aboutToRemoveOrInsert = false;
 
     endRemoveRows();
